Metalama.Framework/Advising/IAdviceFactory.cs

```diff
@@ -781,6 +781,10 @@ public interface IAdviceFactory
             TypeKind typeKind = TypeKind.Class,
```

Comment:
I think the API is wrong here. For example `IntroduceClass(..., TypeKind.Interface)` doesn't make sense. There should either be `IntroduceClass`, etc. with no `TypeKind` parameter, or `IntroduceType` with that parameter, or both.

Reply:
Indeed there should be no TypeKind.